From a036d172e04e78b85bfd7076f8815107746fea87 Mon Sep 17 00:00:00 2001
From: Andy Grover <agrover@redhat.com>
Date: Mon, 19 Aug 2013 17:33:20 -0700
Subject: [PATCH 14/17] iser: Don't wait until iser_ib_init to init list_heads

iser_ib_init exits before initializing list heads if no IB present. This
will cause assert to fire.

see https://bugzilla.redhat.com/show_bug.cgi?id=995750

Signed-off-by: Andy Grover <agrover@redhat.com>
Signed-off-by: FUJITA Tomonori <fujita.tomonori@lab.ntt.co.jp>
---
 usr/iscsi/iser.c |   10 +++-------
 1 files changed, 3 insertions(+), 7 deletions(-)

diff --git a/usr/iscsi/iser.c b/usr/iscsi/iser.c
index a03bb7a..42802be 100644
--- a/usr/iscsi/iser.c
+++ b/usr/iscsi/iser.c
@@ -56,13 +56,13 @@ static struct rdma_event_channel *rdma_evt_channel;
 static struct rdma_cm_id *cma_listen_id;
 
 /* accepted at RDMA layer, but not yet established */
-static struct list_head temp_conn;
+static LIST_HEAD(temp_conn);
 
 /* all devices */
-static struct list_head iser_dev_list;
+static LIST_HEAD(iser_dev_list);
 
 /* all iser connections */
-static struct list_head iser_conn_list;
+static LIST_HEAD(iser_conn_list);
 
 #define uint64_from_ptr(p) (uint64_t)(uintptr_t)(p)
 #define ptr_from_int64(p) (void *)(unsigned long)(p)
@@ -3373,10 +3373,6 @@ static int iser_ib_init(void)
 	if (err)
 		return err;
 
-	INIT_LIST_HEAD(&iser_dev_list);
-	INIT_LIST_HEAD(&iser_conn_list);
-	INIT_LIST_HEAD(&temp_conn);
-
 	return err;
 }
 
-- 
1.7.1

